lgc/patch/VertexFetch.cpp

```diff
@@ -475,6 +506,438 @@ bool LowerVertexFetch::runImpl(Module &module, PipelineState *pipelineState) {
+    Type *argTypes[] = {
+        FixedVectorType::get(builder.getInt32Ty(), 4), // vbdesc
+        builder.getInt32Ty(),                          // vbIndex
+        builder.getInt32Ty(),                          // vertexOffset
+        builder.getInt32Ty(),                          // component size in byte
+        builder.getInt1Ty(),                           // isPacked
+        builder.getInt1Ty(),                           // isBgr
+        builder.getInt1Ty(),                           // Y component mask
+        builder.getInt1Ty(),                           // Z component mask
+        builder.getInt1Ty(),                           // W component mask
+    };
+    // Return type
+    Type *retTy = FixedVectorType::get(builder.getInt32Ty(), is64bitFetch ? 8 : 4);
+
+    StringRef funcName = is64bitFetch ? "FetchVertex64" : "FetchVertex32";
+    FunctionType *const funcTy = FunctionType::get(retTy, argTypes, false);
+    Function *func = Function::Create(funcTy, GlobalValue::InternalLinkage, funcName, module);
+    func->setCallingConv(CallingConv::C);
+    func->addFnAttr(Attribute::AlwaysInline);
+
+    auto entry = createBlock(".entry", func);
+    auto wholeVertex = createBlock(".wholeVertex", func);
+    auto comp0Block = createBlock(".comp0Block", func);
+    auto comp1Block = createBlock(".comp1Block", func);
+    auto comp2Block = createBlock(".comp2Block", func);
+    auto comp3Block = createBlock(".comp3Block", func);
+    auto endfun = createBlock(".endfun", func);
+
+    Value *args[] = {
+        func->getArg(0),     // rsrc
+        func->getArg(1),     // vindex
+        func->getArg(2),     // offset
+        builder.getInt32(0), // soffset
+        builder.getInt32(0)  // glc, slc
+    };
+
+    // .entry
+    {
+      builder.SetInsertPoint(entry);
+      // If ispacked is false, we require per-component fetch
+      builder.CreateCondBr(func->getArg(4), wholeVertex, comp0Block);
+    }
+
+    // .wholeVertex
+    {
+      builder.SetInsertPoint(wholeVertex);
+      Value *vertex = builder.CreateIntrinsic(Intrinsic::amdgcn_struct_buffer_load_format, fetchType, args, {});
+      if (is64bitFetch) {
+        // If it is 64-bit, we need the second fetch
+        args[2] = builder.CreateAdd(args[2], builder.getInt32(SizeOfVec4));
+        auto secondFetch = builder.CreateIntrinsic(Intrinsic::amdgcn_struct_buffer_load_format, fetchType, args, {});
+        std::vector<Constant *> shuffleMask;
+        for (unsigned i = 0; i < 8; ++i)
+          shuffleMask.push_back(ConstantInt::get(Type::getInt32Ty(*m_context), i));
+        vertex = builder.CreateShuffleVector(vertex, secondFetch, ConstantVector::get(shuffleMask));
+      }
+      builder.CreateRet(vertex);
+    }
+
+    // return value
+    Value *lastVert = UndefValue::get(retTy);
+    Value *comp0 = nullptr;
+    Value *comp1 = nullptr;
+    Value *comp2 = nullptr;
+    Value *comp3 = nullptr;
+    // Per-component fetch
+    // reset
+    args[2] = func->getArg(2);
```

Comment:
Please define local variables for capturing the function arguments with meaningful names.